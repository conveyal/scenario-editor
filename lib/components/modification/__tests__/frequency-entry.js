/* global describe, it, expect, jest */

describe('Component > FrequencyEntry', () => {
  const renderer = require('react-test-renderer')
  const React = require('react')
  const {mockFeed, mockTimetable} = require('../../../utils/mock-data')
  const FrequencyEntry = require('../frequency-entry')

  it('renders correctly', () => {
    const replaceTimetableFn = jest.fn()
    const removeTimetableFn = jest.fn()
    const setActiveTripsFn = jest.fn()
    const tree = renderer.create(
      <FrequencyEntry
        feed={mockFeed}
        update={replaceTimetableFn}
        remove={removeTimetableFn}
        setActiveTrips={setActiveTripsFn}
        timetable={mockTimetable}
        routes={['route1']}
        trip='abcd'
        />
    ).toJSON()
    expect(tree).toMatchSnapshot()
    expect(replaceTimetableFn).not.toBeCalled()
    expect(removeTimetableFn).not.toBeCalled()
    expect(setActiveTripsFn).not.toBeCalled()
  })
})
