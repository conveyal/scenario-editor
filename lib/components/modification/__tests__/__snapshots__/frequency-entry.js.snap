// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > FrequencyEntry renders correctly 1`] = `
<section
  className="panel panel-default inner-panel"
  onFocus={[Function]}
>
  <a
    className="panel-heading clearfix"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
    tabIndex={0}
  >
    <i
      className="fa fa-caret-down fa-fw "
    />
    <strong>
       
      Test timetable
    </strong>
  </a>
  <div
    className="panel-body"
  >
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id="name-input-0"
        name="Name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value="Test timetable"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="Patterns"
      >
        Select patterns
      </label>
      <div
        className="Select Select--multi is-searchable"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-2--value"
          >
            <div
              className="Select-placeholder"
            >
              Select patterns
            </div>
          </span>
          <span
            aria-label="Clear all"
            className="Select-clear-zone"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            title="Clear all"
          >
            <span
              className="Select-clear"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&times;",
                }
              }
            />
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="Select Select--single is-searchable has-value"
        style={undefined}
      >
        <input
          disabled={false}
          name="Trip"
          type="hidden"
          value="abcd"
        />
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-3--value"
          >
            <div
              className="Select-value"
              style={undefined}
              title={undefined}
            >
              <span
                aria-selected="true"
                className="Select-value-label"
                id="react-select-3--value-item"
                role="option"
              >
                The Express , starting 03:26 (21 minute trip)
              </span>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        className="form-inline"
      >
        <div
          className="checkbox"
        >
          <label
            htmlFor="mon-input-1"
          >
            <input
              checked={true}
              id="mon-input-1"
              onChange={[Function]}
              type="checkbox"
            />
             
            Mon
          </label>
        </div>
        <div
          className="checkbox"
        >
          <label
            htmlFor="tue-input-2"
          >
            <input
              checked={true}
              id="tue-input-2"
              onChange={[Function]}
              type="checkbox"
            />
             
            Tue
          </label>
        </div>
        <div
          className="checkbox"
        >
          <label
            htmlFor="wed-input-3"
          >
            <input
              checked={true}
              id="wed-input-3"
              onChange={[Function]}
              type="checkbox"
            />
             
            Wed
          </label>
        </div>
        <div
          className="checkbox"
        >
          <label
            htmlFor="thu-input-4"
          >
            <input
              checked={true}
              id="thu-input-4"
              onChange={[Function]}
              type="checkbox"
            />
             
            Thu
          </label>
        </div>
        <div
          className="checkbox"
        >
          <label
            htmlFor="fri-input-5"
          >
            <input
              checked={true}
              id="fri-input-5"
              onChange={[Function]}
              type="checkbox"
            />
             
            Fri
          </label>
        </div>
        <div
          className="checkbox"
        >
          <label
            htmlFor="sat-input-6"
          >
            <input
              checked={false}
              id="sat-input-6"
              onChange={[Function]}
              type="checkbox"
            />
             
            Sat
          </label>
        </div>
        <div
          className="checkbox"
        >
          <label
            htmlFor="sun-input-7"
          >
            <input
              checked={false}
              id="sun-input-7"
              onChange={[Function]}
              type="checkbox"
            />
             
            Sun
          </label>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="frequency-input-8"
        >
          Frequency
        </label>
        <div
          className="InputWithUnits"
        >
          <input
            className="form-control"
            id="frequency-input-8"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="mm:ss"
            type="text"
            value="15:00"
          />
          <span
            className="InputUnits"
          >
            mm:ss
          </span>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="start-time-input-9"
        >
          Start time
        </label>
        <div
          className="InputWithUnits"
        >
          <input
            className="form-control"
            id="start-time-input-9"
            name={undefined}
            onChange={[Function]}
            onWheel={[Function]}
            placeholder="hh:mm"
            type="time"
            value="08:00"
          />
          <span
            className="InputUnits"
          >
            hh:mm
          </span>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="end-time-input-10"
        >
          End time
        </label>
        <div
          className="InputWithUnits"
        >
          <input
            className="form-control"
            id="end-time-input-10"
            name={undefined}
            onChange={[Function]}
            onWheel={[Function]}
            placeholder="hh:mm"
            type="time"
            value="16:00"
          />
          <span
            className="InputUnits"
          >
            hh:mm
          </span>
        </div>
      </div>
      <div
        className="checkbox"
      >
        <label
          htmlFor="times-are-exact-input-11"
        >
          <input
            checked={undefined}
            disabled={false}
            id="times-are-exact-input-11"
            onChange={[Function]}
            type="checkbox"
          />
           
          Times are exact
        </label>
      </div>
      <div>
        <div
          className="form-group"
        >
          <label
            htmlFor="phase-at-stop-input-12"
          >
            Phase at stop
          </label>
          <div
            className="Select Select--single is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-4--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Phase at stop
                </div>
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <a
      className="btn btn-danger btn-block"
      href="#"
      onClick={[Function]}
      tabIndex={0}
      title="Delete frequency entry"
    >
      <i
        className="fa fa-close fa-fw "
      />
       Delete frequency entry
    </a>
  </div>
</section>
`;
